src/Microsoft.ML/Data/CollectionDataSource.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+
+namespace Microsoft.ML.Data
+{
+    public static class CollectionDataSource
```

Comment:
>CollectionDataSource [](start = 24, length = 20)

Please add a top level XML help - what is this class, where does it use, what's the purpose for its existance?